lib/LedgerSMB/Admin/Command/user.pm

```diff
@@ -0,0 +1,582 @@
+package LedgerSMB::Admin::Command::user;
+
+=head1 NAME
+
+LedgerSMB::Admin::Command::user - ledgersmb-admin 'user' command
+
+=cut
+
+use strict;
+use warnings;
+
+use Getopt::Long qw(GetOptionsFromArray);
+use LedgerSMB::Admin::Command;
+use LedgerSMB::App_State;
+use LedgerSMB::Company;
+use LedgerSMB::Database;
+use LedgerSMB::DBObject::User;
+use LedgerSMB::Entity::Person::Employee;
+use LedgerSMB::Entity::User;
+use LedgerSMB::PGDate;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+use Feature::Compat::Try;
+
+# LedgerSMB database interface
+has db => (is => 'rw');
+has super_dbh => (is => 'rw',
+                  predicate => '_has_super_dbh');
+has admin_dbh => (is => 'rw', lazy => 1,
+                  builder => '_build_admin_dbh',
+                  clearer => '_clear_admin_dbh',
+                  predicate => '_has_admin_dbh',
+    );
+
+# users
+my $username;
+my $password;
+
+# employee
+my $start_date; # date
+my $end_date; # date
+my $dob; # date
+my $role; # text
+my $ssn; # text 
+my $is_sales; # bool 
+my $manager; # text
+my $employeenumber; # text
+my $is_manager; # bool
+# RETURNS id integer AS $$
+
+# person
+my $salutation = ''; # Default to none
+my $first_name; # text 
+my $middle_name; # text 
+my $last_name; # text
+my $country; # text
+#my $personal_id; # text
+my $permission; # Text
+my $no_permission; # Text
+
+sub _get_valid_salutation {
+    my ($self, $dbh) = @_;
+    my $sth = $dbh->prepare('SELECT * FROM salutation');
+    $sth->execute
+        or die $dbh->errstr;
+    my ($values) = $sth->fetchall_arrayref({});
+    foreach (@$values) {
+        return $_->{id}
+            if ( $_->{salutation} eq $salutation)
+    }
+    $self->logger->error('Invalid salutation');
+    return 0;
+}
+
+sub _get_user {
+    my ($self, $dbh) = @_;
+    my $_username = shift // $username;
+
+    # The code below doesn't provide the entity_id
+    #my $user_obj = LedgerSMB::DBObject::User->new();
+    #$user_obj->set_dbh($dbh);
+    #my @users = @{$user_obj->get_all_users};
+    #my ($user) = grep { $_username eq $_->{username} } @users;
+
+    #This ugly hack does
+    my $sth = $dbh->prepare("SELECT id, entity_id FROM users WHERE username='$username'");
```

Comment:
Please use placeholders and not string interpolation with SQL commands.

Reply:
Done wherever possible.